corehq/apps/locations/migrations/0009_populate_location_type_from_string.py

```
@@ -0,0 +1,162 @@
+# encoding: utf-8
+from south.v2 import DataMigration
+from dimagi.utils.decorators.memoized import memoized
+from dimagi.utils.couch.database import get_db, iter_docs
+from dimagi.utils.chunked import chunked
+
+
+EXCLUDE_DOMAINS = (
+    "drewpsi",
+    "psi",
+    "psi-ors",
+    "psi-test",
+    "psi-test2",
+    "psi-test3",
+    "psi-unicef",
+    "psi-unicef-wb",
+)
+
+
+class Migration(DataMigration):
+
+    @memoized
+    def domain_loc_types(self, domain):
+        domain = get_db().view("domain/domains", key=domain, reduce=False,
+                               include_docs=True).one()
+        return {
+            # It looks like code is what I should use, not name
+            loc_type['code']: loc_type
+            for loc_type in domain.get('location_types', [])
+        }
+
+    def iter_relevant_locations(self):
+        return (self.orm.SQLLocation.objects.exclude(domain__in=EXCLUDE_DOMAINS)
+                                       .iterator())
+
+    @memoized
+    def get_loc_type(self, domain, code):
+        """
+        Get or create relevant SQL location type
+        """
+        try:
+            loc_type = self.orm.LocationType.objects.get(domain=domain, code=code)
+        except self.orm.LocationType.DoesNotExist:
+            loc_types = self.domain_loc_types(domain)
+            if not loc_types:
```

if no location types in the domain this should fail louder